### PR TITLE
Faction Selector GUI

### PR DESCRIPTION
**Pull request by killerswin2**
## What type of PR is this.
1. [ ] Bug
2. [ ] Change
3. [ ] Enhancement

### What have you changed and why?
Information:
    THIS IS FOR HAKON AND FROST TO LOSE THERE MIND

### Please specify which Issue this PR Resolves.
closes #XXXX

### Please verify the following and ensure all checks are completed.

1. [ ] Have you loaded the mission in LAN host?
2. [ ] Have you loaded the mission on a dedicated server?

### Is further testing or are further changes required?
1. [ ] No
2. [ ] Yes (Please provide further detail below.)

### How can the changes be tested?
Steps: 

********************************************************
Notes:

### CODE REVIEW

**Review by HakonRydland — COMMENTED** [review]
ToDo:
- Add addon selection
- Add information to info panel
- Add support for seeing mod-sets lacking addons
    - Add info as tooltips to what mod(s) are lacking with icons if we can
    
 - Consider moving to controlGroups if need be to make adding the lacking features easier/possible

**Comment by killerswin2** [comment]
Screenshot 2022-04-10 104930

**Comment by killerswin2** [comment]
Need to save the selection, as to bypass the selector again.

File: A3A/addons/GUI/functions/GUI/fn_factionSelectorDialog.sqf [hunk 1]
```diff
@@ -0,0 +1,278 @@
...
+	} forEach _factionList;
+}forEach _civTemplates;
+
+// do the nodes
+private _vanillaNodes = _nodes findIf {"Vanilla" in _x};
+call compile preprocessFileLineNumbers (_nodes deleteAt _vanillaNodes);
+{
+    call compile preprocessFileLineNumbers _x;
+    Info_1("Loading logistic nodes: %1", _x);
+} forEach _nodes;
```
Comment (HakonRydland) [inline A3A/addons/GUI/functions/GUI/fn_factionSelectorDialog.sqf:278]: the selector addaptation is a bit sloopy but i think i can do the heavy lifiting in cleaning it up if you want

**Review by HakonRydland — CHANGES_REQUESTED** [review]
for the most part good
- liked the use of inheritance to cut down on copy paste
- liked the use of pixel grid

what needs to be fixed (not including afformentioned stuff from a few days ago)
- localisation
- selector template parsing (i can do this if you want)
- addon selection for UI
- template info section

as mentioned before if youd like to consider going the dynamic list of controls group you can compress information into teh list itself along with having tooltips etc for eatch entry, this is considerably harder though and if youd rather not we can live with it being deligated to the current info window

**Pull request merged**